flake.nix

```diff
@@ -59,57 +59,13 @@
     // flake-utils.lib.eachDefaultSystem (
       system: let
         lib = pkgs.callPackage ./lib/lib.nix {
-          inherit system;
           flakeRoot = self.outPath;
           nickel = inputs.nickel.packages."${system}".nickel-lang-cli;
-          organistLib = self.lib.${system};
         };
         pkgs = nixpkgs.legacyPackages.${system};
       in {
         lib.importNcl = lib.importNcl;
 
-        # Helper function that generates contents for "nickel.lock.ncl", see buildLockFile.
-        lib.buildLockFileContents = contents: let
-          lib = pkgs.lib;
-          getLinesOne = indent: name: thing:
-            if lib.isAttrs thing
-            then
-              [
-                (indent + (lib.optionalString (name != null) "${name} = ") + "{")
-              ]
-              ++ lib.mapAttrsToList (getLinesOne (indent + "  ")) thing
-              ++ [
-                (indent + "}" + (lib.optionalString (name != null) ","))
-              ]
-            else [''${indent}${name} = import "${builtins.toString thing}",''];
-        in
-          lib.concatLines (lib.flatten (getLinesOne "" null contents));
-
-        # A script that generates contents of "nickel.lock.ncl" file from a recursive attribute set of strings.
-        # Example inputs:
-        #   {
-        #     organist = {
-        #       builders = "/nix/store/...-source/builders.ncl";
-        #       contracts = "/nix/store/...-source/contracts.ncl";
-        #       nix = "/nix/store/...-source/nix.ncl";
-        #     };
-        #   }
-        # Result:
-        #   {
-        #     organist = {
-        #       builders = import "/nix/store/...-source/builders.ncl",
-        #       contracts = import "/nix/store/...-source/contracts.ncl",
-        #       nix = import "/nix/store/...-source/nix.ncl",
-        #     },
-        #   }
-        lib.buildLockFile = contents:
-          pkgs.writeShellApplication {
-            name = "regenerate-lockfile";
-            text = ''
-              cat > nickel.lock.ncl <${builtins.toFile "nickel.lock.ncl" (self.lib.${system}.buildLockFileContents contents)}
-            '';
-          };
-
         # Flake app to generate nickel.lock.ncl file. Example usage:
```

Comment:
I think we shouldn't stop there and move `regenerateLockFileApp` also to `lib/lib.nix` and simplify this flake file by just calling `lib = pkgs.callPackage lib/lib.nix {...}`. Then also pieces that were already moved to `lib/lib.nix` will still be accessible via flake.